airbyte-integrations/connectors/source-hubspot/source_hubspot/manifest.yaml

```diff
@@ -508,13 +508,277 @@ definitions:
+    retriever:
+      $ref: "#/definitions/base_retriever"
+      requester:
+        $ref: "#/definitions/base_requester"
+        path: /email/public/v1/events
+      record_selector:
+        schema_normalization: Default
+        extractor:
+          type: DpathExtractor
+          field_path:
+            - events
+      paginator:
+        type: DefaultPaginator
+        page_token_option:
+          type: RequestOption
+          inject_into: request_parameter
+          field_name: offset
+        page_size_option:
+          type: RequestOption
+          inject_into: request_parameter
+          field_name: limit
+        pagination_strategy:
+          type: CursorPagination
+          page_size: 1000
+          cursor_value: "{{ response.get('offset') }}"
+          stop_condition: "{{ not response.get('hasMore') }}"
+    incremental_sync:
+      type: DatetimeBasedCursor
+      cursor_field: created
+      datetime_format: "%ms"
+      lookback_window: P{{ config.get('lookback_window', 0) }}D
+      step: P30D
+      cursor_granularity: PT1S
+      cursor_datetime_formats:
+        - "%ms"
+        - "%Y-%m-%dT%H:%M:%S.%fZ"
+        - "%Y-%m-%dT%H:%M:%SZ"
+      start_datetime:
+        type: MinMaxDatetime
+        datetime: "{{ format_datetime(config.get('start_date', '2006-06-01T00:00:00Z'), '%ms', '%Y-%m-%dT%H:%M:%SZ') }}"
+        datetime_format: "%ms"
+      start_time_option:
+        type: RequestOption
+        field_name: startTimestamp
+        inject_into: request_parameter
+      end_time_option:
+        type: RequestOption
+        field_name: endTimestamp
+        inject_into: request_parameter
+    schema_loader:
+      type: InlineSchemaLoader
+      schema:
+        $ref: "#/schemas/email_events"
+
+  engagements_stream:
+    $ref: "#/definitions/stream_base"
+    name: engagements
+    primary_key: id
+    retriever:
+      $ref: "#/definitions/base_retriever"
+      requester:
+        type: CustomRequester
+        class_name: source_hubspot.components.EngagementsHttpRequester
+        url_base: https://api.hubapi.com
+        authenticator:
+          $ref: "#/definitions/authenticator"
+        http_method: GET
+        request_headers:
+          Content-Type: application/json
+          User-Agent: Airbyte
+        error_handler:
+          $ref: "#/definitions/base_error_handler"
+      record_selector:
+        type: RecordSelector
+        schema_normalization: Default
+        transform_before_filtering: true
+        extractor:
+          type: DpathExtractor
+          field_path:
+            - results
+      paginator:
+        type: DefaultPaginator
+        page_token_option:
+          type: RequestOption
+          inject_into: request_parameter
+          field_name: offset
+        page_size_option:
+          type: RequestOption
+          inject_into: request_parameter
+          field_name: count
+        pagination_strategy:
+          type: CursorPagination
+          page_size: 250
```

Comment:
The `/recent` endpoint actually only allows for a max of 100 items per page according to their documentation: https://developers.hubspot.com/docs/reference/api/crm/engagements/engagement-details/v1#get-recent-engagements

Can you confirm that this request still works when `250` is specified? I assume that it just returns 100, but we should make sure. And assuming that this still works, we should add a comment mentioning that even though we specify `250`, the `/recent` endpoint will still return 100 records and not throw an error despite being over the limit